### PR TITLE
improve RestrictedEndpointsAdmission test

### PR DESCRIPTION
**Pull request by danwinship**
Updates the RestrictedEndpointAdmissions e2e test to test EndpointSlices as well.

Requires https://github.com/openshift/kubernetes/pull/899

### CODE REVIEW

**Comment by danwinship** [comment]
/hold
because it's waiting for https://github.com/openshift/kubernetes/pull/899
but it can be reviewed now

**Comment by danwinship** [comment]
/retest
/assign @astoycos 
since you've looked at a bunch of e2e stuff...

File: test/extended/networking/endpoint_admission.go [hunk 1]
```diff
@@ -75,6 +77,50 @@ var _ = g.Describe("[sig-network][endpoints] admission", func() {
...
+		testOneEndpointSlice(oc, serviceAccountClient, "external", "1.2.3.4", true)
+
+		// Project admin
+		testOneEndpointSlice(oc, projectAdminClient, "cluster", clusterIP, false)
+		testOneEndpointSlice(oc, projectAdminClient, "service", serviceIP, false)
+		// EndpointSlice is more locked-down than Endpoints, so even this will fail...
+		testOneEndpointSlice(oc, projectAdminClient, "external", "1.2.3.4", false)
+
+		// Now try giving the project admin endpointslice edit permission...
+		out, err := oc.AsAdmin().Run("create", "role", "--namespace", oc.Namespace(), "endpointslice-edit", "--verb=update", "--resource=endpoints.discovery.k8s.io").Output()
```
Comment (zhaozhanqi) [inline test/extended/networking/endpoint_admission.go:102]: s/endpoints/endpointslices/ ?

Comment (danwinship) [reply test/extended/networking/endpoint_admission.go]: uh... yes...

**Comment by astoycos** [comment]
Nice work super readable :)

**Comment by danwinship** [comment]
/hold cancel
@astoycos can you re-lgtm? I had to [re-push](https://github.com/openshift/origin/compare/3b674dbaf44ffc03785fe9b31afa67eaba8ebf4a..90919d7adc345a214300fdc0789257086f289db3) because... uh... it didn't work :slightly_smiling_face:

**Comment by astoycos** [comment]
Ah I probably should have made sure the test was actually working.. Looks like it's passing now though :+1:

**Pull request merged**